crates/starknet-devnet-server/src/api/json_rpc/mod.rs

```diff
@@ -83,15 +83,15 @@ pub struct JsonRpcHandler {
 impl RpcHandler for JsonRpcHandler {
     type Request = StarknetRequest;
 
-    async fn on_request(&self, request: Self::Request) -> ResponseResult {
+    async fn on_request(&mut self, request: Self::Request) -> ResponseResult {
```

Comment:
why the reference is being changed to mut

Reply:
I was sure that it didn't compile without `mut`; looks like it's compiling. Addressed in https://github.com/0xSpaceShard/starknet-devnet-rs/pull/329/commits/1999c72cc0f02d2e098c826899446d00952f11ab